resources/public/js/cluster.js

```diff
@@ -1546,25 +1546,27 @@ function Cluster() {
       recoveryListing.append('<li role="separator" class="divider"></li>');
 
       // Suggest successor
-      instance.children.forEach(function(replica) {
-        if (!replica.LogBinEnabled) {
-          return
-        }
-        if (replica.SQLDelay > 0) {
-          return
-        }
-        if (!replica.LogReplicationUpdatesEnabled) {
-          return
-        }
-        if (replica.lastCheckInvalidProblem()) {
-          return
-        }
-        if (replica.notRecentlyCheckedProblem()) {
-          return
-        }
-        recoveryListing.append(
-          '<li><a href="#" data-btn="recover-suggested-successor" data-command="recover-suggested-successor" data-successor-host="' + replica.Key.Hostname + '" data-successor-port="' + replica.Key.Port + '">Recover, try to promote <code>' + replica.title + '</code></a></li>');
-      });
+      if (instance.children) {
+        instance.children.forEach(function(replica) {
+          if (!replica.LogBinEnabled) {
```

Comment:
these conditions can be merged into multiple OR statements

Reply:
Agree, but this was the original, upstream code, so it is desired to keep it with minimal changes.